memory/src/columns.rs

```diff
@@ -27,31 +28,29 @@ pub struct MemoryCols<T> {
 
     /// A boolean flag indicating whether addr' - addr == 0
     pub addr_not_equal: T,
+
+    pub counter: T,
```

Comment:
I didn't understand what these represent?

Reply:
This is to keep track of the table length needed for the range check. I still need to add constraints to check that this is monotonically increasing by one.